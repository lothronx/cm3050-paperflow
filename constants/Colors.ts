export const COLORS = {
  primary: "#6A7BFF",
  primaryActive: "#5A67FF",
  secondary: "#F07067",
  secondaryActive: "#D65A67",
  background: "#FFFFFF",
  backgroundSecondary: "#F5F5F5",
  text: "#2D3436",
  textSecondary: "#636E72",
  border: "#DFE6E9",
};
