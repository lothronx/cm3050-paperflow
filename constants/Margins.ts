import { Platform } from "react-native";

// iOS needs a bigger vertical margin because of its safe area display feature
export const MARGINS = {
  top: Platform.OS === "ios" ? 64 : 24,
  horizontal: 18,
  bottom: Platform.OS === "ios" ? 32 : 24,
};
