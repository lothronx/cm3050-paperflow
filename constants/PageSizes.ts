export const PageSizes = {
  A4: {
    width: 595,
    height: 842,
  },
  Letter: {
    width: 612,
    height: 792,
  },
  Legal: {
    width: 612,
    height: 1008,
  },
} as const;