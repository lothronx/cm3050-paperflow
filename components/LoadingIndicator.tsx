import { View, ActivityIndicator, StyleSheet } from "react-native";
import { COLORS } from "@/constants/Colors";

export function LoadingIndicator() {
  return (
    <View style={styles.loadingContainer} testID="loading-container">
      <ActivityIndicator size="large" color={COLORS.primary} testID="loading-indicator" />
    </View>
  );
}

const styles = StyleSheet.create({
  loadingContainer: {
    position: "absolute",
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    zIndex: 100,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "rgba(0, 0, 0, 0.25)",
  },
});
