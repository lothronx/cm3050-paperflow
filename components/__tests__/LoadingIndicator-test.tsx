import React from 'react';
import { render } from '@testing-library/react-native';
import { LoadingIndicator } from '../LoadingIndicator';
import { COLORS } from '@/constants/Colors';

describe('<LoadingIndicator />', () => {
  test('renders correctly', () => {
    const { getByTestId } = render(<LoadingIndicator />);
    
    // Check if ActivityIndicator is present
    const activityIndicator = getByTestId('loading-indicator');
    expect(activityIndicator).toBeTruthy();
  });

  test('matches snapshot', () => {
    const tree = render(<LoadingIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('has correct styling', () => {
    const { getByTestId } = render(<LoadingIndicator />);
    
    // Get the container view
    const container = getByTestId('loading-container');
    
    // Check if the container has the correct styles
    expect(container.props.style).toMatchObject({
      position: 'absolute',
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      zIndex: 100,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: 'rgba(0, 0, 0, 0.25)',
    });
  });

  test('ActivityIndicator has correct props', () => {
    const { getByTestId } = render(<LoadingIndicator />);
    
    // Get the ActivityIndicator
    const activityIndicator = getByTestId('loading-indicator');
    
    // Check if it has the correct size and color
    expect(activityIndicator.props.size).toBe('large');
    expect(activityIndicator.props.color).toBe(COLORS.primary);
  });
});