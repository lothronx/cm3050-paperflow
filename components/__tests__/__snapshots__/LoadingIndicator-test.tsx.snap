// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.25)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 100,
    }
  }
  testID="loading-container"
>
  <ActivityIndicator
    color="#6A7BFF"
    size="large"
    testID="loading-indicator"
  />
</View>
`;
